File: engine/engine-model/src/main/scala/tools/packet/monster/carnival/MonsterCarnivalPointObtained.scala
package tools.packet.monster.carnival

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class MonsterCarnivalPointObtained(private var _currentPoints: Int, private var _totalPoints: Int) extends PacketInput {
  def currentPoints: Int = _currentPoints

  def totalPoints: Int = _totalPoints

  override def opcode(): SendOpcode = SendOpcode.MONSTER_CARNIVAL_OBTAINED_CP
}

File: engine/engine-model/src/main/java/tools/packet/monster/carnival/MonsterCarnivalPointObtained.java
package tools.packet.monster.carnival;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record MonsterCarnivalPointObtained(Integer currentPoints, Integer totalPoints) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MONSTER_CARNIVAL_OBTAINED_CP;
   }
}